### PR TITLE
[KYUUBI #5404] JDBC Engine supports StarRocks

### LINKED ISSUE #5404: [TASK][EASY] JDBC Engine supports StarRocks
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before creating

- [X] I have searched in the [task list](https://github.com/orgs/apache/projects/296) and found no similar tasks.

### Mentor

- [ ] I have sufficient knowledge and experience of this task, and I volunteer to be the mentor of this task to guide contributors to complete the task.

### Skill requirements

* Basic knowledge of Scala programming language
* Basic knowledge of JDBC concepts
* Basic knowledge of Docker and [Testcontianers](https://java.testcontainers.org/)

### Background and Goals

Currently, Kyuubi supports JDBC engines with limited dialects, and we can extend the dialects to support [StarRocks](https://www.starrocks.io/), which is an open-source and high-performance analytical database.

Reference https://www.bilibili.com/video/BV1UT411m7LR

### Implementation steps

* Implement a StarRocks dialect in the JDBC engine
* Leverage the Testcontainers to add integration tests
* Update document correspondingly

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bowenliang123**
# :mag: Description
## Issue References 🔗

This pull request fixes #5404

## Describe Your Solution 🔧

- Introduce StarRocks support in JDBC Engine
- Adding dialects and tests for StarRocks
- Tested with StarRocks 3.x official open-sourced version docker image, with a mini cluster with FE and BE setup.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Behavior With This Pull Request :tada:
Supported StarRocks in JDBC engine.

#### Related Unit Tests
Add tests of `StarRocksOperationSuite`, `StarRocksOperationWithEngineSuite`, `StarRocksSessionSuite` and `StarRocksStatementSuite`.

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [ ] Pull request title is okay.
- [ ] No license issues.
- [ ] Milestone correctly set?
- [ ] Test coverage is ok
- [ ] Assignees are selected.
- [ ] Minimum number of approvals
- [ ] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

File: externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/mysql/Mysql8ConnectionProvider.scala [hunk 1]
```diff
@@ -22,10 +22,14 @@ class Mysql8ConnectionProvider extends JdbcConnectionProvider {
...
-  override val driverClass: String = "com.mysql.cj.jdbc.Driver"
+  override val driverClass: String = Mysql8ConnectionProvider.driverClass
 
   override def canHandle(providerClass: String): Boolean = {
     driverClass.equalsIgnoreCase(providerClass)
   }
 
 }
+
+object Mysql8ConnectionProvider {
```
Comment (pan3793) [inline externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/mysql/Mysql8ConnectionProvider.scala:33]:
```suggestion
object MySQL8ConnectionProvider {
```

Comment (bowenliang123) [reply externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/mysql/Mysql8ConnectionProvider.scala]: Fixed.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -2905,6 +2905,8 @@ object KyuubiConf {
           "org.apache.kyuubi.engine.jdbc.phoenix.PhoenixConnectionProvider"
         case "PostgreSQL" | "postgresql" | "PostgreSQLConnectionProvider" =>
           "org.apache.kyuubi.engine.jdbc.postgresql.PostgreSQLConnectionProvider"
+        case "StarRocks" | "starrocks" | "StarRocksConnectionProvider" =>
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:2908]: the doc should be updated as well

**Review by pan3793 — APPROVED** [review]

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 2 above)
Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:2909]: Thanks for the reminders. Updated.

**Comment by bowenliang123** [comment]
Thanks, merged to master(1.9.0).

**Pull request closed without merging**